precompiles/ArbAggregator.go

```diff
@@ -20,87 +16,111 @@ type ArbAggregator struct {
+	if !isOwner {
+		return ErrNotOwner
+	}
+	batchPosterTable := c.State.L1PricingState().BatchPosterTable()
+	isBatchPoster, err := batchPosterTable.ContainsPoster(newDefault)
 	if err != nil {
 		return err
 	}
-	if !allowed {
-		return errors.New("Only the current default (or its fee collector / chain owner) may change the default")
+	if !isBatchPoster {
+		_, err = batchPosterTable.AddPoster(newDefault, newDefault)
+		if err != nil {
+			return err
+		}
 	}
-	return l1State.SetDefaultAggregator(newDefault)
-}
-
-// Get the aggregator's compression ratio, measured in basis points
-func (con ArbAggregator) GetCompressionRatio(c ctx, evm mech, aggregator addr) (uint64, error) {
-	ratio, err := c.State.L1PricingState().AggregatorCompressionRatio(aggregator)
-	return uint64(ratio), err
+	return nil
 }
 
-// Set the aggregator's compression ratio, measured in basis points
-func (con ArbAggregator) SetCompressionRatio(c ctx, evm mech, aggregator addr, newRatio uint64) error {
-	allowed, err := accountIsAggregatorOrCollectorOrOwner(c.caller, aggregator, c.State)
+func (con ArbAggregator) AddBatchPoster(c ctx, evm mech, newBatchPoster addr) error {
+	isOwner, err := c.State.ChainOwners().IsMember(c.caller)
 	if err != nil {
 		return err
 	}
-	if !allowed {
-		return errors.New("Only an aggregator (or its fee collector / chain owner) may change its compression ratio")
+	if !isOwner {
+		return ErrNotOwner
 	}
-	return c.State.L1PricingState().SetAggregatorCompressionRatio(aggregator, arbmath.Bips(newRatio))
+	batchPosterTable := c.State.L1PricingState().BatchPosterTable()
+	isBatchPoster, err := batchPosterTable.ContainsPoster(newBatchPoster)
+	if err != nil {
+		return err
+	}
+	if !isBatchPoster {
+		_, err = batchPosterTable.AddPoster(newBatchPoster, newBatchPoster)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
-// Gets an aggregator's fee collector
-func (con ArbAggregator) GetFeeCollector(c ctx, evm mech, aggregator addr) (addr, error) {
-	return c.State.L1PricingState().AggregatorFeeCollector(aggregator)
+// Gets a batch poster's fee collector
+func (con ArbAggregator) GetFeeCollector(c ctx, evm mech, batchPoster addr) (addr, error) {
+	posterInfo, err := c.State.L1PricingState().BatchPosterTable().OpenPoster(batchPoster)
+	if err != nil {
+		return addr{}, err
+	}
+	return posterInfo.PayTo()
 }
 
-// Sets an aggregator's fee collector (caller must be the aggregator, its fee collector, or an owner)
-func (con ArbAggregator) SetFeeCollector(c ctx, evm mech, aggregator addr, newFeeCollector addr) error {
-	allowed, err := accountIsAggregatorOrCollectorOrOwner(c.caller, aggregator, c.State)
+// Sets a batch poster's fee collector (caller must be the batch poster, its fee collector, or an owner)
+func (con ArbAggregator) SetFeeCollector(c ctx, evm mech, batchPoster addr, newFeeCollector addr) error {
+	posterInfo, err := c.State.L1PricingState().BatchPosterTable().OpenPoster(batchPoster)
```

Comment:
As discussed, we should prevent the creation of a batch poster here

Reply:
Fixed